### ISSUE TITLE
mdBook Implementation

### ISSUE THREAD

**Issue by wilwade** [LibertyDSNP/spec#144]
Our static site generator isn't really meeting our needs currently. From #141 to some of the issues with docz.

To Do:
- [x] Build out mdBook Implementation (see https://github.com/LibertyDSNP/spec/tree/spec_mdBook branch for help)
  - [x] Update Menu to be better nested
  - [x] Configure Menu to collapse
  - [x] Use the default theme
- [x] Update the README with how to and getting started
- [x] Figure out a way to still support our linting (either by switching linters or configuring the current ones better)
- [x] Switch OFF GitHub Actions Deployment
- [x] Enable some of the https://github.com/rust-lang/mdBook/wiki/Third-party-plugins that appear useful

### Acceptance
- [x] Can locally follow instructions in the README to run the spec site
- [x] Can see the spec
- [x] Can lint the spec

Previous Notes:
We thought about each of these:

- https://rust-lang.github.io/mdBook/ which we tried out https://github.com/LibertyDSNP/spec/tree/spec_mdBook
- https://gohugo.io/ (Used for example here: https://spec.filecoin.io/)

tldr so far is:
- The default style of mdBook and some of the themes with Hugo are both fairly close to what we would want.
- mdBook is simpler and is Rust based so super fast, but also a bit harder to use our current linting tools. Has the backing of the Rust community.
- Hugo has more plugins and options, but is slower to build. Has a larger community.

**Comment by wilwade** [comment]
Finish trying out mdBook and see how far it gets us.

**Comment by mehtaishita** [comment]
There's a missing section (content-proof) in the Attachments page right after the Video Link table.  
See /pages/ActivityContent/Associated/Attachments.md for [content proofs section] link
